app/src/main/java/fr/free/nrw/commons/settings/SettingsFragment.java

```diff
@@ -144,10 +141,42 @@ public void afterTextChanged(Editable s) {
         }
     }
 
+    /**
+     * Uses previously saved theme if there is any, if not then uses default.
+     * Adds preference theme listener and saves value chosen by user to shared preferences
+     * to remember later
+     */
+    private void prepareTheme() {
+
+        String currentTheme = getCurrentTheme();
+        if (currentTheme.equals("")){
+            // If current theme code is empty, means none selected by user yet so use default
+            themeListPreference.setSummary(getString(R.string.theme_default_name));
+            themeListPreference.setValue(getString(R.string.theme_default_value));
+        } else {
+            // If any theme is selected by user previously, use it
+            int prefIndex = themeListPreference.findIndexOfValue(currentTheme);
+            themeListPreference.setSummary(themeListPreference.getEntries()[prefIndex]);
```

Comment:
You can just use `simpleSummary`, see [the docs](https://developer.android.com/guide/topics/ui/settings/customize-your-settings#dynamically_update_summaries)

Reply:
I am unable to use simpleSumarry, I followed the instructions from the docs but there is no method **ListPreference.SimpleSummaryProvider.getInstance()** even after adding **app:useSimpleSummaryProvider="true"** to XML

Reply:
Okay _now_ I care about us not using the androidx artifacts. So irritating.

How do you feel about a follow up ticket to this? `Change Preferences to use AndroidX`?

Reply:
@macgills Sure, I think we should do it in a separate PR,

Reply:
Absolutely, I'll make it and assign you, cheers.